codebundles/k8s-kubectl-namespace-healthcheck/sli.robot

```diff
@@ -0,0 +1,114 @@
+*** Settings ***
+Metadata          Author    Shea Stewart
+Documentation     This SLI uses kubectl to score namespace health. Produces a value between 0 (completely failing thet test) and 1 (fully passying the test). Looks for container restarts, events, and pods not ready.
+Force Tags        K8s    Kubernetes    Kube    K8    Namespace    Health
+Suite Setup       Suite Initialization
+Library           BuiltIn
+Library           RW.Core
+Library           RW.Utils
+Library           RW.K8s
+Library           RW.platform
+Library           OperatingSystem
+
+*** Keywords ***
+Suite Initialization
+    ${kubeconfig}=    RW.Core.Import Secret    kubeconfig
+    ...    type=string
+    ...    description=The kubernetes kubeconfig yaml containing connection configuration used to connect to cluster(s).
+    ...    pattern=\w*
+    ...    example=For examples, start here https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/
+    ${kubectl}=    RW.Core.Import Service    kubectl
+    ...    description=The location service used to interpret shell commands.
+    ...    default=kubectl-service.shared
+    ...    example=kubectl-service.shared
+    ${NAMESPACE}=    RW.Core.Import User Variable    NAMESPACE
+    ...    type=string
+    ...    description=The name of the Kubernetes namespace to scope actions and searching to. Supports csv list of namespaces, or ALL. 
+    ...    pattern=\w*
+    ...    example=ALL
+    ${CONTEXT}=    RW.Core.Import User Variable    CONTEXT
+    ...    type=string
+    ...    description=Which Kubernetes context to operate within.
+    ...    pattern=\w*
+    ...    example=my-main-cluster
+    ${DISTRIBUTION}=    RW.Core.Import User Variable    DISTRIBUTION
+    ...    type=string
+    ...    description=Which distribution of Kubernetes to use for operations, such as: Kubernetes, OpenShift, etc.
+    ...    pattern=\w*
+    ...    enum=[Kubernetes,GKE,OpenShift]
+    ...    example=Kubernetes
+    ${EVENT_TYPE}=    RW.Core.Import User Variable    EVENT_TYPE
+    ...    type=string
+    ...    description=The error pattern to use when grep-ing logs or searching events.
+    ...    pattern=\w*
+    ...    example=Normal|Warning
+    ...    default=Warning
+    ${EVENT_AGE}=    RW.Core.Import User Variable    EVENT_AGE
+    ...    type=string
+    ...    description=The time window in minutes as to when the event was last seen.
+    ...    pattern=\w*
```

Comment:
Depending on time units you intended to accept, something like `    ...    pattern=((\d+?)d)?((\d+?)h)?((\d+?)m)?((\d+?)s)?`

Reply:
See previous comment, I think we should just stick to minutes for now.

Reply:
modified the function to support wider time ranges, but locked the form input to `minutes` only as events are only stored for 1 h.